apps/extensions_manager/lib/extensions_manager/model_extension.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ExtensionsManager.ModelExtension do
+  defmacro __using__(_opts) do
+    quote do
+      Module.register_attribute(__MODULE__, :schema_changes, accumulate: true)
+      Module.register_attribute(__MODULE__, :method_block, accumulate: true)
+
+      import ExtensionsManager.ModelExtension, only: [add_to_schema: 4, add_to_schema: 3, include_method: 1]
+      @before_compile ExtensionsManager.ModelExtension
+
+      defmacro __using__(_opts) do
```

Comment:
Does defining a `__using__/1` prevent the user from defining their own `__using__/1`? Since most `use X` statements are at the top, that'd mean this one would match everything and any other `__using__/1` definitions would generate a compiler warning and never match.